app/controllers/InstanceRegistryController.scala

```diff
@@ -68,7 +76,195 @@ class InstanceRegistryController @Inject()(myExecutionContext: MyExecutionContex
+      ){
+        val obj = jsonValue.as[JsObject]
+        println(obj)
+        if((obj \ "param").asOpt[String].get.equals("InstanceID")){
+          instanceID = (obj \ "value").asOpt[String].get.substring(1)
+          //instanceID = toInt(intanceID)
+          println("winnie test send parameter-->>" +instanceID)
+        }
+      }
+    }
+      println(instanceRegistryUri)
+    ws.url(instanceRegistryUri + "/start")
+      .addQueryStringParameters("Id" -> instanceID)
+      .post("")
+      .map { response =>
+        response.status match {
+          case 202 =>
+            println("winnie start with id" +response.body)
+            Ok(response.body)
+          case x =>
+            println("winnie error" +response)
+            new Status(x)
+        }
+      }(myExecutionContext)
+  }
+
+  def stopInstance() : Action[AnyContent] = Action.async { request =>
+    var instanceID = ""
+    request.body.asJson.map { json =>
+      for(jsonValue <- (json \ "params" \ "updates").as[JsValue].as[Seq[JsValue]]
+      ){
+        val obj = jsonValue.as[JsObject]
+        println(obj)
+        if((obj \ "param").asOpt[String].get.equals("InstanceID")){
+          instanceID = (obj \ "value").asOpt[String].get.substring(1)
+          //instanceID = toInt(intanceID)
+          println("winnie test send parameter-->>" +instanceID)
+        }
+      }
+    }
+    println(instanceRegistryUri)
+    ws.url(instanceRegistryUri + "/stop")
+      .addQueryStringParameters("Id" -> instanceID)
+      .post("")
+      .map { response =>
+        response.status match {
+          case 202 =>
+            println("winnie stop with id" +response.body)
+            Ok(response.body)
+          case x =>
+            println("winnie stop error" +response)
+            new Status(x)
+        }
+      }(myExecutionContext)
+  }
+
+  def pauseInstance() : Action[AnyContent] = Action.async { request =>
+    var instanceID = ""
+    request.body.asJson.map { json =>
+      for(jsonValue <- (json \ "params" \ "updates").as[JsValue].as[Seq[JsValue]]
+      ){
+        val obj = jsonValue.as[JsObject]
+        println(obj)
+        if((obj \ "param").asOpt[String].get.equals("InstanceID")){
+          instanceID = (obj \ "value").asOpt[String].get.substring(1)
+          //instanceID = toInt(intanceID)
+          println("winnie test send parameter-->>" +instanceID)
+        }
+      }
+    }
+    println(instanceRegistryUri)
+    ws.url(instanceRegistryUri + "/pause")
+      .addQueryStringParameters("Id" -> instanceID)
+      .post("")
+      .map { response =>
+        response.status match {
+          case 202 =>
+            println("winnie pause with id" +response.body)
+            Ok(response.body)
+          case x =>
+            println("winnie pause error" +response)
+            new Status(x)
+        }
+      }(myExecutionContext)
+  }
+
+  def resumeInstance() : Action[AnyContent] = Action.async { request =>
```

Comment:
please find a way to reuse the code of the methods handling the post requests. Currently all methods contain basically the same code only derivating in the given parameters. Try to create a data model describing the general type of request, parse this data model and create a universally useable method handling this data model and creating the post request to the instance registry

Reply:
Created a data model in the latest commit which can handle all the POST requests.